### ISSUE TITLE
Expanding arguments for WHERE IN queries does not work in Android

### ISSUE THREAD

**Issue by knaeckeKami** [simolus3/moor#156]
There seems to be a platform difference between Android and iOS on how query parameters in WHERE IN (...) queries are handled.

Example:
db.dart: https://gist.github.com/knaeckeKami/299969459246f01b51862d67c3c986b0
main.dart: https://gist.github.com/knaeckeKami/8fafa26fab6bf830e0457f4566dc71de

Output on Android:

```
I/flutter ( 7194): Moor: Sent CREATE TABLE IF NOT EXISTS dummy_table (id INTEGER NOT NULL); with args []
I/flutter ( 7194): Moor: Executing [BatchedStatement(INSERT INTO dummy_table (id) VALUES (?), [[1], [2], [3]])] in a batch
I/flutter ( 7194): Moor: Sent select * from dummy_table where id in (?1, ?2, ?3) with args [1, 2, 3]
I/flutter ( 7194): **i don't work**
I/flutter ( 7194): Moor: Sent select * from dummy_table where id in (?,?,?) with args [1, 2, 3]
I/flutter ( 7194): **i work**
```

Output on iOS:

```
Syncing files to device iPhone Xʀ...
flutter: Moor: Sent CREATE TABLE IF NOT EXISTS dummy_table (id INTEGER NOT NULL); with args []
flutter: Moor: Executing [BatchedStatement(INSERT INTO dummy_table (id) VALUES (?), [[1], [2], [3]])] in a batch
flutter: Moor: Sent select * from dummy_table where id in (?1, ?2, ?3) with args [1, 2, 3]
flutter: **i work**
flutter: Moor: Sent select * from dummy_table where id in (?,?,?) with args [1, 2, 3]
flutter: **i work**
```

It seems that numbered host parameters do not work in this context on android for some reason. 

Avoiding numbered parameters and just using "?" as placeholder seems to be a workaround.

**Comment by simolus3** [comment]
Yikes. This is a bug in sqflite (see https://github.com/tekartik/sqflite/issues/272). Sqflite inlines Integer parameters into the query because the underlying Android API is acting up.

**Comment by simolus3** [comment]
This is fixed with `sqflite` version [`1.1.6+5`](https://pub.dev/packages/sqflite#1165). Running `flutter packages uprade` should fix this problem. To be sure that you're never using an affected version, I would recommend to add a direct dependency on `sqflite: ^1.1.6+5` when using `moor_flutter`. The next `moor_flutter` version will have that version as a lower bound.

I've also added an integration test case for this. We already had some tests to verify that queries expand correctly, but they use a mocked database. I'll try to run more unit tests as integration test across multiple database implementations in the future.
